integration-tests/src/test/java/com/datastax/oss/driver/api/core/compression/DirectCompressionIT.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2017-2017 DataStax Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.compression;
+
+import com.datastax.oss.driver.api.core.Cluster;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.api.core.cql.SimpleStatement;
+import com.datastax.oss.driver.api.core.session.Session;
+import com.datastax.oss.driver.api.testinfra.ccm.CcmRule;
+import com.datastax.oss.driver.api.testinfra.cluster.ClusterRule;
+import com.datastax.oss.driver.api.testinfra.cluster.ClusterUtils;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.offset;
+
+public class DirectCompressionIT {
+
+  @ClassRule public static CcmRule ccmRule = CcmRule.getInstance();
+
+  @ClassRule
+  public static ClusterRule schemaClusterRule =
+      new ClusterRule(ccmRule, "request.timeout = 30 seconds");
+
+  @BeforeClass
+  public static void setup() {
+    schemaClusterRule
+        .session()
+        .execute("CREATE TABLE test (k text PRIMARY KEY, t text, i int, f float)");
+  }
+
+  /**
+   * Validates that a cluster configured with Snappy compression and can execute queries that insert
+   * and retrieve data.
+   *
+   * @test_category connection:compression
+   * @expected_result session established and queries made successfully using it.
+   */
+  @Test
+  public void should_execute_queries_with_snappy_compression() throws Exception {
+    createAndCheckCluster(
+        "protocol.compressor.class = "
+            + "com.datastax.oss.driver.internal.core.protocol.SnappyCompressor");
+  }
+
+  /**
+   * Validates that a cluster configured with LZ4 compression and can execute queries that insert
+   * and retrieve data.
+   *
+   * @test_category connection:compression
+   * @expected_result session established and queries made successfully using it.
+   */
+  @Test
+  public void should_execute_queries_with_lz4_compression() throws Exception {
+    createAndCheckCluster(
+        "protocol.compressor.class = "
+            + "com.datastax.oss.driver.internal.core.protocol.Lz4Compressor");
+  }
+
+  private void createAndCheckCluster(String compressorOption) {
+
+    try (Cluster cluster =
+        ClusterUtils.newCluster(ccmRule, compressorOption, "request.timeout = 30 seconds")) {
```

Comment:
The request.timeout isn't needed here right?  That's more needed for making schema changes I had thought.